orderer/consensus/follower/follower_chain.go

```diff
@@ -51,16 +143,128 @@ func (*Chain) Errored() <-chan struct{} {
 }
 
 func (c *Chain) Start() {
-	//TODO skeleton
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.started || c.stopped {
+		c.logger.Debugf("Not starting because: started=%v, stopped=%v", c.started, c.stopped)
+		return
+	}
+
+	c.started = true
+
+	go c.run()
+
+	c.logger.Info("Started")
 }
 
 func (c *Chain) Halt() {
-	//TODO skeleton
+	c.halt()
+
+	select {
+	case <-c.doneChan:
+	}
+}
+
+func (c *Chain) halt() {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if c.stopped {
+		c.logger.Debug("Already stopped")
+		return
+	}
+	c.stopped = true
+	close(c.stopChan)
+	c.logger.Info("Stopped")
 }
 
-// StatusReport returns the ClusterRelation & Status
+// StatusReport returns the ClusterRelation & Status.
 func (c *Chain) StatusReport() (types.ClusterRelation, types.Status) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	clusterRelation := types.ClusterRelationFollower
+	if c.joinBlock != nil {
+		if err := c.amIInChannel(c.joinBlock); err == nil {
+			clusterRelation = types.ClusterRelationMember
+		}
+	}
+
 	status := types.StatusActive
-	//TODO if (height is >= join-block.height) return "active"; else return "onboarding"
-	return types.ClusterRelationFollower, status
+	if (c.joinBlock != nil) && (c.joinBlock.Header.Number >= c.support.Height()) {
+		status = types.StatusOnBoarding
+	}
+
+	return clusterRelation, status
+}
+
+func (c *Chain) run() {
+	c.logger.Debug("The follower.Chain puller goroutine is starting")
+	pullRetryTicker := time.NewTicker(500 * time.Millisecond) // TODO move constant to options
+
+	err := c.pull()
+
+runLoop:
+	for err != nil {
```

Comment:
I really dislike labels and goto :( 

an alternative simpler solution would be:
```

defer func() {
	pullRetryTicker.Stop()
	close(c.doneChan)
	c.logger.Debug("The follower.Chain puller goroutine is exiting")
}

for {
  select {
    case   <- pullRetryTicker.C:
       // do nothing
   case <- stopChain:
           return
    if err = c.pull(); err != nil {
           return
    }
}
```

Reply:
What you wrote in the example does not work. When the ticker hits, we want to pull again, not do nothing, and when we pull and get error we want to try again after the next ticker hits, not return. 

However I got your intention, one can certainly put the tail of the function in a defer and return early when the stop signal hits. 

And for the record, the "tags" are encouraged in go for[ nested structures that use break](https://forum.golangbridge.org/t/is-using-continue-label-or-break-label-good-practice/8345) , but I agree it is redundant here.

Reply:
I put the bracket in the wrong place... I updated the example.

Reply:
I understand how to eliminate the tag... 
In your corrected example the first pull happens only after a tick... and the return needs to be `if err==nil`...
But again, got your intention and will correct it.

Reply:
Done